modules/consumableBidAdapter.js

```diff
@@ -0,0 +1,199 @@
+    let dict = values[values.length - 1] || {};
+    let result = [strings[0]];
+    keys.forEach(function(key, i) {
+      let value = isInteger(key) ? values[key] : dict[key];
+      result.push(value, strings[i + 1]);
+    });
+    return result.join('');
+  };
+}
+
+function parsePixelItems(pixels) {
+  let itemsRegExp = /(img|iframe)[\s\S]*?src\s*=\s*("|')(.*?)\2/gi;
+  let tagNameRegExp = /\w*(?=\s)/;
+  let srcRegExp = /src=("|')(.*?)\1/;
+  let pixelsItems = [];
+
+  if (pixels) {
+    let matchedItems = pixels.match(itemsRegExp);
+    if (matchedItems) {
+      matchedItems.forEach(item => {
+        let tagName = item.match(tagNameRegExp)[0];
+        let url = item.match(srcRegExp)[2];
+
+        if (tagName && tagName) {
+          pixelsItems.push({
+            type: tagName === SYNC_TYPES.IMAGE.TAG ? SYNC_TYPES.IMAGE.TYPE : SYNC_TYPES.IFRAME.TYPE,
+            url: url
+          });
+        }
+      });
+    }
+  }
+
+  return pixelsItems;
+}
+
+function _buildConsumableUrl(bid) {
+  const params = bid.params;
+
+  let server = CONSUMABLE_URL;
+  params.network = params.network || '10947.1'
+
+  return pubapiTemplate({
+    host: server,
+    network: params.network,
+    placement: parseInt(params.placement),
+    misc: new Date().getTime() // cache busting
+  });
+}
+
+function formatBidRequest(bid) {
+  let bidRequest;
+
+  bidRequest = {
+    url: _buildConsumableUrl(bid),
+    method: 'GET',
+    ttl: CONSUMABLE_TTL
+  };
+
+  bidRequest.bidderCode = bid.bidder;
+  bidRequest.bidId = bid.bidId;
+  bidRequest.userSyncOn = bid.params.userSyncOn;
+  bidRequest.unitId = bid.params.unitId;
+  bidRequest.unitName = bid.params.unitName;
+  bidRequest.zoneId = bid.params.zoneId;
+  bidRequest.network = bid.params.network;
+
+  return bidRequest;
+}
+
+export const spec = {
+  code: CONSUMABLE_BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    return bid.params && bid.params.placement
+  },
+  buildRequests: function (bids) {
+    return bids.map(bid => {
+      return formatBidRequest(bid);
+    });
+  },
+  interpretResponse: function ({body}, bidRequest) {
+    if (!body) {
+      utils.logError('Empty bid response', bidRequest.bidderCode, body);
+    } else {
+      let bid = this._parseBidResponse(body, bidRequest);
+      if (bid) {
+        return bid;
+      }
+    }
+  },
+  _formatPixels: function (pixels) {
+    let formattedPixels = pixels.replace(/<\/?script( type=('|")text\/javascript('|")|)?>/g, '');
+
+    return '<script>var w=window,prebid;' +
+      'for(var i=0;i<10;i++){w = w.parent;prebid=w.$$PREBID_GLOBAL$$;' +
+      'if(prebid && prebid.consumableGlobals && !prebid.consumableGlobals.pixelsDropped){' +
+      'try{prebid.consumableGlobals.pixelsDropped=true;' + formattedPixels + 'break;}' +
+      'catch(e){continue;}' +
+      '}}</script>';
+  },
+  _parseBidResponse: function (response, bidRequest) {
+    let bidData;
+    try {
+      bidData = response.seatbid[0].bid[0];
```

Comment:
If your endpoint is returning arrays (which it appears to be by this code) I suggest handling the responses as arrays w/ `forEach` or `map` (if you're returning the results directly) rather than picking off the first bid and bailing early if not present.  `interpretResponse` allows you to return multiple bids.  Even if you only want to return one bid for now, a `forEach` or `map` implementation will still behave correctly (as you have here) but will also be forward compatible should you decide to do multiple bids in the future.

Reply:
Good suggestion. I'd like to wait on this to roll it into a larger refactor. It should work fine as-is as we're only concerned with the first bid in the array returned.